icicle-source/src/Icicle/Source/Checker/Invariants.hs

```diff
@@ -48,16 +48,13 @@ invariantQ ctx (Query (c:cs) xfinal)
      -> errBanLatest
 
     GroupBy _ x
-     -> goX x >> goBanWindowAndGroupFold
+     -> goX x >> go
 
     Distinct _ x
-     -> goX x >> goBanWindowAndGroupFold
+     -> goX x >> go
 
     GroupFold _ _ _ x
-     | allowWindows inv
-     -> goX x >> goBanWindowAndGroupFold
-     | otherwise
-     -> errBanGroupFold
```

Comment:
the change actually made it less strict already, from the group in group change: https://github.com/ambiata/icicle/commit/88901cd08b51f4988913c000142ee8f74e5f616e#diff-743f2c2eb9de71059af5941e74e658c6